scripts/createDB.sql

```diff
@@ -31,302 +33,326 @@ CREATE TABLE `soen_390_db`.`customer_order` (
   PRIMARY KEY (`customer_order_id`))
   
   collate = utf8mb4_unicode_ci;
+
+-- Table to store the different orders of raw materials needed for producing bikes.
+CREATE TABLE `soen_390_db`.`raw_material_order` (
+  `raw_material_order_id` INT NOT NULL AUTO_INCREMENT,
+  `status` VARCHAR(45) NOT NULL,
+  `date_time_added` DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  `date_time_finished` DATETIME,
+  `price` DECIMAL(10,2) NOT NULL,
+  PRIMARY KEY (`raw_material_order_id`))
   
+  collate = utf8mb4_unicode_ci;
+
+-- Table to store the different orders given by the manufacturing division
+-- in order to create various semi-finished and finished goods.
 CREATE TABLE `soen_390_db`.`production_order` (
   `production_order_id` INT NOT NULL AUTO_INCREMENT,
   `status` VARCHAR(45) NOT NULL,
   `date_time_added` DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
   `date_time_finished` DATETIME,
-  `price` DECIMAL(10,2) NOT NULL,
   PRIMARY KEY (`production_order_id`))
   
   collate = utf8mb4_unicode_ci;
 
+-- Table to store the raw materials, semi-finished goods, and finished goods all
+-- under one table.
+CREATE TABLE `soen_390_db`.`inventory_item` (
+  `inventory_item_id` INT NOT NULL AUTO_INCREMENT,
+  `inventory_item_name` VARCHAR(45) NOT NULL,
+  `inventory_item_type` VARCHAR(45) NOT NULL,
+  `quantity` INT NOT NULL DEFAULT 0,
+  `date_time_added` DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  PRIMARY KEY (`inventory_item_id`),
+  CONSTRAINT `validItemType`
+    CHECK (`inventory_item_type` IN ("raw", "semi-finished", "finished")))
+  
+  collate = utf8mb4_unicode_ci;
+
+CREATE TABLE `soen_390_db`.`inventory_item_composed_of` (
+  `inventory_item_id` INT NOT NULL,
+  `made_from_inventory_item_id` INT NOT NULL,
+  `quantity` INT NOT NULL,
+  PRIMARY KEY (`inventory_item_id`, `made_from_inventory_item_id`),
+  CONSTRAINT `inventoryItemIDComposedOfForeignKey`
+    FOREIGN KEY (`inventory_item_id`)
+    REFERENCES `soen_390_db`.`inventory_item` (`inventory_item_id`)
+    ON DELETE CASCADE
+    ON UPDATE NO ACTION,
+  CONSTRAINT `madeFromInventoryItemIDComposedOfForeignKey`
+    FOREIGN KEY (`made_from_inventory_item_id`)
+    REFERENCES `soen_390_db`.`inventory_item` (`inventory_item_id`)
+    ON DELETE NO ACTION
+    ON UPDATE NO ACTION,
+  CONSTRAINT `notMadeFromSameItem`
+    CHECK (`inventory_item_id` != `made_from_inventory_item_id`))
+    
+	collate = utf8mb4_unicode_ci;
+
+-- Table to store the raw materials needed to build the semi-finished goods.
 CREATE TABLE `soen_390_db`.`raw_material` (
```

Comment:
We should change the name raw material to raw good that way we have finished_good, semi_finished_good and raw_good